drivers/sdhc/sdhc_spi.c

```diff
@@ -148,24 +149,30 @@ static int sdhc_spi_init_card(const struct device *dev)
 	const struct sdhc_spi_config *config = dev->config;
 	struct sdhc_spi_data *data = dev->data;
 	struct spi_config *spi_cfg = data->spi_cfg;
-	int ret;
+	int ret, ret2;
 
 	if (spi_cfg->frequency == 0) {
 		/* Use default 400KHZ frequency */
 		spi_cfg->frequency = SDMMC_CLOCK_400KHZ;
 	}
+
+	/* Request SPI bus to be active */
+	if (pm_device_runtime_get(config->spi_dev) < 0) {
+		return -EIO;
+	}
+
 	/* the initial 74 clocks must be sent while CS is high */
 	spi_cfg->operation |= SPI_CS_ACTIVE_HIGH;
 	ret = sdhc_spi_rx(config->spi_dev, spi_cfg, data->scratch, 10);
-	if (ret != 0) {
-		spi_release(config->spi_dev, spi_cfg);
-		spi_cfg->operation &= ~SPI_CS_ACTIVE_HIGH;
-		return ret;
-	}
+
 	/* Release lock on SPI bus */
-	ret = spi_release(config->spi_dev, spi_cfg);
+	ret2 = spi_release(config->spi_dev, spi_cfg);
```

Comment:
Why is the result of spi_release() relevant here at all?

Reply:
Because it was being returned previously? Not sure why it was introduced, I just kept it.

Reply:
IMO, the construct looks confusing, but I have no better suggestion.